raft/src/main/java/org/apache/kafka/raft/FileBasedStateStore.java

```diff
@@ -149,6 +149,7 @@ private void writeElectionStateToFile(final File stateFile, QuorumStateData stat
             writer.write(jsonState.toString());
             writer.flush();
             fileOutputStream.getFD().sync();
+            fileOutputStream.close();
```

Comment:
Should we close writer before close fileOutputStream? if we should should close writer first, I think we can also get rid of "try with resource" here.

Reply:
I would move `Utils.atomicMoveWithFallback` out of the _try with resource_:
```
	try {
		try (FileOutputStream fileOutputStream = ...) {
		         ...
		}
		Utils.atomicMoveWithFallback(temp.toPath(), file.toPath());
	} catch (final IOException e) {
```

That way neither a comment nor a duplicate close is needed.

Reply:
@liefke's suggestion is also my preference. @renaldobf are you able to continue with the PR?

Reply:
I created this PR which includes this suggestion: https://github.com/apache/kafka/pull/14354

cc @dengziming @showuon